src/Database/Query.php

```diff
@@ -315,7 +316,7 @@ public function sql(?ValueBinder $generator = null): string
      * @param callable $visitor A function or callable to be executed for each part
      * @return $this
      */
-    public function traverse(callable $visitor)
+    public function traverse($visitor)
```

Comment:
Had to remove type hint and rely on parameter widening here as `Query implements ExpressionInterface` and `ExpressionInterface::traverse()` only accepts `Closure` now.

Reply:
Is the concern that userland code is using `traverse()` on a query object?

Reply:
Yes, lorenzo commented [here](https://github.com/cakephp/cakephp/pull/13611#issuecomment-527997495) that the accepted type shouldn't be narrowed down for query methods like formatResults, traverse, map... etc.